File: client/GhostClient/Core/Network/Packets/MapSettingsPacket.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using GhostClient.Core;
using Microsoft.Xna.Framework;

namespace Ghost.Core.Network.Packets
{
    public class MapSettingsPacket : Packet
    {
        public MapSettingsPacket(Stream stream) : base(stream)
        {
        }

        public MapSettingsPacket(Stream stream, bool autoClose) : base(stream, autoClose)
        {
        }

        public MapSettingsPacket(byte[] data) : base(data)
        {
        }

        protected override void OnHandlePacket()
        {
            float power = Consume(4).FloatValue;

            int red = Consume(4).IntValue;
            int green = Consume(4).IntValue;
            int blue = Consume(4).IntValue;

            int mapNameLength = Consume(4).IntValue;
            string mapName = Consume(mapNameLength).AsciiStringValue;

            Color color = Color.FromNonPremultiplied(red, green, blue, 255);
            GhostClient.Ghost.AmbientPower = power;
            GhostClient.Ghost.AmbientColor = color;

            GameHandler.Game.PrepareMap(mapName);
        }
    }
}

File: client/core/src/com/boxtrotstudio/ghost/client/network/packets/MapSettingsPacket.kt
package com.boxtrotstudio.ghost.client.network.packets

import com.badlogic.gdx.graphics.Color
import com.boxtrotstudio.ghost.client.Ghost
import com.boxtrotstudio.ghost.client.network.Packet
import com.boxtrotstudio.ghost.client.network.PlayerClient

class MapSettingsPacket : Packet<PlayerClient>() {

    override fun handle() {
        val power = consume(4).asFloat()

        val red = consume(4).asInt()
        val green = consume(4).asInt()
        val blue = consume(4).asInt()

        val mapNameLength = consume(4).asInt()
        val mapName = consume(mapNameLength).asString()

        Ghost.rayHandler.ambientLight = Color((red / 255f) * power, (green / 255f) * power, (blue / 255f) * power, 1f)

        client.game.prepareMap(mapName)
    }
}
